Tests/webcomponents/js/ez-universal-discovery.js

```diff
@@ -0,0 +1,409 @@
+/* global describe, it, beforeEach, fixture, assert, afterEach, sinon */
+describe('ez-universal-discovery', function () {
+    let universalDiscovery;
+    let universalDiscoveryMultiple;
+    const modelObject = 'hasBeenTransformedToModelObject';
+
+    function getAppMock() {
+        const App = function () {
+            this.renderSideView = function () {};
+        };
+
+        return new App();
+    }
+
+    function getViewMock(domNode) {
+        const View = function () {
+            this.set = function () {};
+            this.get = function () {
+                return {
+                    getDOMNode: function () {
+                        return domNode;
+                    },
+                };
+            };
+            this.destroy = function () {};
+        };
+
+        return new View();
+    }
+
+    function defineGlobals() {
+        window.eZ = {
+            YUI: {
+                app: getAppMock(),
+                Y: {
+                    eZ: {
+                        UniversalDiscoveryView: function () {},
+                        UniversalDiscoveryViewService: function () {},
+                    }
+                }
+            }
+        };
+    }
+
+    function getModelMock() {
+        const Model = function () {
+            this.toObject = function () {
+                return modelObject;
+            };
+        };
+
+        return new Model();
+    }
+
+    function dispatchEvent(handlerName, selection) {
```

Comment:
that's a very confusing name. OK it dispatches an event (`ez:yui-app:ready`) but it's not really the point of it. It should rahter be called `runUDWEventHandler` or something like that.